### ISSUE TITLE
Add support for most commons HTTP errors 

### ISSUE THREAD

**Issue by nomorepanic** [slimphp/Slim#1965]
Hi everyone! I have been recently using Slim to build a ReST API.

 Slim already provides handlers for 404, 405 and 500 errors, but to my understanding handlers set in this way are used internally by Slim.

If I wanted to return an error, currently, I'd have to interact with the response

```
return $response->withStatus(400)->withJson(/*some message*/);
```

What could happen instead is:

```
return Slim\Http\Errors::badRequest('my message')
```

This is only an example and  is just some syntactic sugar. For some errors it may also actually be useful, such as in case of 401 Unathorized where an header is also required.

**Comment by akrabat** [comment]
Please explain more?

**Comment by nomorepanic** [comment]
Sorry about that, accidentally pressed enter :S

**Comment by dopesong** [comment]
So you can use custom exception handler and throw some kind of exception while your custom handler catches it and does what you want to do. :)

**Comment by geggleto** [comment]
Maybe for 4.x

**Comment by akrabat** [comment]
Implemented in #2398
